eventsources/sources/amqp/start.go

```diff
@@ -106,7 +106,11 @@ func (el *EventListener) StartListening(ctx context.Context, dispatch func([]byt
 	log.Info("listening to messages on channel...")
 	for {
 		select {
-		case msg := <-delivery:
+		case msg, ok := <-delivery:
+			if !ok {
+				log.Error("failed to read a message, channel might have been closed")
+				return errors.New("channel might have been closed")
```

Comment:
Will have retries in eventing.go.

Reply:
@whynowy thanks for your effort! 
Can you give me the PR for retries, so i can follow it?

Reply:
https://github.com/argoproj/argo-events/pull/927